Src/Support/GoogleApis.Auth.DotNet4/OAuth2/AuthorizationDialog.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using Google.Apis.Auth.OAuth2.Responses;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    public partial class AuthorizationDialog : Form
+    {
+        private AuthorizationCodeResponseUrl responseUrl;
+
+        public AuthorizationCodeResponseUrl ResponseUrl
+        {
+            get
+            {
+                return responseUrl;
+            }
+        }
+        /// <summary> Constructs a new authentication broker dialog. </summary>
+        public AuthorizationDialog(Uri authUri)
+        {
+            InitializeComponent();
+            // Set the browser to the initial uri.
+            authWebBrowser.Navigate(authUri);
+            // Need to track when the results has been received.
+            authWebBrowser.DocumentCompleted += authWebBrowser_DocumentCompleted;
```

Comment:
> Yes, adding any UI dependency to the DLL would have given me the same concern. IMO the client lib should not be for UI it should just be for the background functionality of accessing the APIs.

As a counterargument, there is already precedent for including UI elements in the library. The Windows Phone platform includes UI controls to present the browser to the user from within the library. The references to System.Windows.Controls are UI dependencies.